core/src/main/java/io/questdb/TelemetryJob.java

```diff
@@ -0,0 +1,174 @@
+import io.questdb.mp.*;
+import io.questdb.std.*;
+import io.questdb.std.microtime.*;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+import io.questdb.tasks.TelemetryTask;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.Closeable;
+
+public class TelemetryJob extends SynchronizedJob implements Closeable {
+    private static final Log LOG = LogFactory.getLog(TelemetryJob.class);
+
+    private final CharSequence tableName = "telemetry";
+    private final CharSequence configTableName = "telemetry_config";
+    private final QueueConsumer<TelemetryTask> myConsumer = this::newRowConsumer;
+    private final StringSink idSink = new StringSink();
+
+    private final MicrosecondClock clock;
+    private final PropServerConfiguration configuration;
+    private final boolean enabled;
+    private final TableWriter writer;
+    private final TableWriter writerConfig;
+    private final RingQueue<TelemetryTask> queue;
+    private final SCSequence subSeq;
+
+    public TelemetryJob(PropServerConfiguration configuration, CairoEngine engine, @Nullable MessageBus messageBus) throws SqlException, CairoException {
+        final CairoConfiguration cairoConfig = configuration.getCairoConfiguration();
+
+        this.clock = cairoConfig.getMicrosecondClock();
+        this.configuration = configuration;
+        this.enabled = configuration.getTelemetryConfiguration().getEnabled();
+        this.queue = messageBus.getTelemetryQueue();
+        this.subSeq = messageBus.getTelemetrySubSequence();
+
+        try (
+                Path path = new Path();
+                SqlCompiler compiler = new SqlCompiler(engine)
+        ) {
+            final SqlExecutionContextImpl sqlExecutionContext = new SqlExecutionContextImpl(messageBus, 1, engine);
+            sqlExecutionContext.with(AllowAllCairoSecurityContext.INSTANCE, null, null);
+
+            if (getTableStatus(path, tableName) == TableUtils.TABLE_DOES_NOT_EXIST) {
+                compiler.compile("CREATE TABLE " + tableName + " (ts timestamp, event short)", sqlExecutionContext);
+            }
+
+            if (getTableStatus(path, configTableName) == TableUtils.TABLE_DOES_NOT_EXIST) {
+                compiler.compile("CREATE TABLE " + configTableName + " (id long256, enabled boolean)", sqlExecutionContext);
+            }
+        }
+
+        try {
+            this.writer = new TableWriter(cairoConfig, tableName);
+        } catch (CairoException ex) {
+            LOG.error().$("could not open [table=").utf8(tableName).$("]").$();
+            throw ex;
+        }
+
+        // todo: close writerConfig. We currently keep it opened to prevent users from modifying the table.
+        // Once we have a permission system, we can use that instead.
+        try {
+            this.writerConfig = new TableWriter(cairoConfig, configTableName);
+        } catch (CairoException ex) {
+            LOG.error().$("could not open [table=").utf8(configTableName).$("]").$();
```

Comment:
if you cannot open "writerConfig", constructor will leak "writer" instance.

Reply:
Done